test_data/rules/unawaited_futures.dart

```diff
@@ -71,6 +71,10 @@ foo12() async {
   x..[0] = fut();
 }
 
+foo13() async {
+  var y = '${fut()}'; //LINT
```

Comment:
An uncommon corner case that we don't need to address in this PR, but should the lint flag in the following location:
```
var a;
var b = '${a = fut()}';
```

Reply:
I think there are test cases against this, e.g. the below doesn't lint

```dart
foo4() async {
  var x = fut();
  return x;
}
```

and the code in

https://github.com/dart-lang/linter/blob/bee643570ba670720b479f0248e28e788faac1a5/lib/src/rules/unawaited_futures.dart#L82